tools/repoctl/src/repoctl/fetchmodels.py

```
@@ -0,0 +1,42 @@
+    if not mdir.is_dir():
+        print(f"fetch-models FAIL: manifest 目录不存在: {mdir}", file=sys.stderr)
+        return 2
+    models = [m for f in sorted(mdir.glob("*.y*ml")) for m in parse_yaml_list(f.read_text())]
+    errs, fails = [], []
+    for m in models:
+        mid, src, sha = m.get("id", ""), m.get("source", ""), (m.get("sha256") or "").lower()
+        u = urlparse(src)
+        spath = Path(u.path if u.scheme == "file" else src)
+        dest = m.get("dest") or spath.name
+        if not mid or not src or len(sha) != 64 or u.scheme not in ("", "file"):
+            errs.append(f"{mid or src}: 清单字段非法或非 file:// 源（本卡桩不下载）")
```

Ruff (0.16.2)

[warning] String contains ambiguous `（` (FULLWIDTH LEFT PARENTHESIS). Did you mean `(` (LEFT PARENTHESIS)?

(RUF001)